### PR TITLE
AIOHttp client gotcha

### PR DESCRIPTION
**Pull request by jianshen92**

### CODE REVIEW

File: aiohttp.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def __call__(self) -> aiohttp.ClientSession:
+        assert self.session is not None
+        return self.session
+```
+
+1. Having a singleton like `HttpClient`, can be used in conjucntion with your web server that makes 3rd party API call. Only initialize the session once at the beginning of the startup
+
+2. `aiohttp.TCPConnector(force_close=True)` will fix the bug where sometimes connection pool is not released after request.
+
+3. `aiohttp.ClientTimeout(total=5)` sets timeout for your requests, in seconds. Response that takes longer that 5 seconds will throw a TimeoutError
```
Comment (keatmin) [inline aiohttp.md:31]: Super cool

File: aiohttp.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    async def stop(self):
+        await self.session.close()
+        self.session = None
+
+    def __call__(self) -> aiohttp.ClientSession:
+        assert self.session is not None
+        return self.session
+```
+
+1. Having a singleton like `HttpClient`, can be used in conjucntion with your web server that makes 3rd party API call. Only initialize the session once at the beginning of the startup
```
Comment (keatmin) [inline aiohttp.md:27]: `conjunction`?

Comment (keatmin) [reply aiohttp.md:27]: I am curious what are the pros and cons of doing so

**Review by keatmin — APPROVED** [review]
lgtm

File: aiohttp.md (hunk 2 above)
Comment (jianshen92) [reply aiohttp.md:27]: not super sure. but initializing objects has a takes more than zero second

**Pull request merged**